src/main/java/org/embulk/input/marketo/delegate/MarketoBaseBulkExtractInputPlugin.java

```diff
@@ -234,7 +235,11 @@ public void jsonColumn(Column column)
 
     private LineDecoderIterator getLineDecoderIterator(T task)
     {
-        List<MarketoUtils.DateRange> dateRanges = MarketoUtils.sliceRange(new DateTime(task.getFromDate()), new DateTime(task.getToDate().orNull()), MARKETO_MAX_RANGE_EXTRACT);
+        final OffsetDateTime fromDate = OffsetDateTime.ofInstant(task.getFromDate().toInstant(), ZoneId.systemDefault());
+        final OffsetDateTime toDate = task.getToDate().isPresent() ?
```

Comment:
toDate value is set at `#validateInputTask` so: no need to check presentation here.

Reply:
I would rely on checks done in other methods, they tend to change time to time.
`new DateTime(task.getToDate().orNull()` and this should be translated in 1to1 manner IMHO